DOCSUP-3908: mannWitneyUTest, welchTTest and studentTTest documentation

lehasm opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation for #10351, #15874, #16883

Detailed description:
Aggregate functions documented: mannWitneyUTest, welchTTest and studentTTest.

Also fixed text messages in source file AggregateFunctionMannWhitney.h:
* parameter value `'two sided'` to  `'two-sided'`
* duplicate words `require require`

Commit:
studentTTest function description

Commit:
Merge branch 'master' of https://github.com/ClickHouse/ClickHouse into alexey-sm-DOCSUP-3908-mannWitneyUTest-welchTTest-studentTTest-doc

lehasm committed:
welchttest description, links in studentttest

lehasm committed:
mannWhitneyUTest description

lehasm committed:
Text changed slightly

lehasm committed:
Text changed slightly

lehasm committed:
Text changed slightly

lehasm committed:
Text changed slightly

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
```diff
@@ -0,0 +1,72 @@
+toc_title: mannWhitneyUTest
+---
+
+## mannWhitneyUTest {#mannwhitneyutest}
```
```suggestion
# mannWhitneyUTest {#mannwhitneyutest}
```

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
```diff
@@ -0,0 +1,72 @@
+    -   `"two-sided"`;
+    -   `"greater"`;
+    -   `"less"`;
+    -   if parameter is not specified then default two sided hypothesis is tested.
```
```suggestion
-   `alternative` — hypothesis tested. [String](../../../sql-reference/data-types/string.md). Possible values:
    -   `"two-sided"` (by default)
    -   `"greater"`
    -   `"less"`
```

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
```diff
@@ -0,0 +1,72 @@
+
+**Parameters**
+
+-   `alternative` — alternative hypothesis. [String](../../../sql-reference/data-types/string.md)
```
```suggestion
-   `alternative` — hypothesis tested. [String](../../../sql-reference/data-types/string.md). Possible values:
    -   `'two-sided'` (by default)
    -   `'greater'`
    -   `'less'`
```

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
```diff
@@ -0,0 +1,72 @@
+**Returned values**
+
+-   statistic - calculated statistic;
+-   p-value - calculated p-value.
```
```suggestion
-  ` statistic` - calculated statistic;
-   `p-value` - calculated p-value.
```

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
```diff
@@ -0,0 +1,72 @@
+
+**Returned values**
+
+-   statistic - calculated statistic;
```
```suggestion
-  ` statistic` - calculated statistic;
```

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
````diff
@@ -0,0 +1,72 @@
+SELECT mannWhitneyUTest('greater')(sample_data, sample_index)
+FROM mww_ttest;
+```
+
````
``` sql
SELECT mannWhitneyUTest('greater')(sample_data, sample_index) FROM mww_ttest;
```

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
```diff
@@ -0,0 +1,72 @@
+## mannWhitneyUTest {#mannwhitneyutest}
+
+Applies the Mann-Whitney rank test to independent samples from distinct distributions.
+
```
```suggestion
Applies the [Mann–Whitney U test](https://en.wikipedia.org/wiki/Mann%E2%80%93Whitney_U_test) to independent samples from distinct distributions.
```

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
```diff
@@ -0,0 +1,72 @@
+
+## mannWhitneyUTest {#mannwhitneyutest}
+
+Applies the Mann-Whitney rank test to independent samples from distinct distributions.
```
```suggestion
Applies the [Mann–Whitney U test](https://en.wikipedia.org/wiki/Mann%E2%80%93Whitney_U_test) to independent samples from distinct distributions.
```

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
```diff
@@ -0,0 +1,72 @@
+Values X and Y are randomly selected from two populations. The null hypothesis is that the probability of X being greater than Y is equal to the probability of Y being greater than X. Also one-sided hypothesis can be tested.
+  
+Values of both samples are in the `sample_data` column. If `sample_index` equals to 0 then the value in that row belongs to the first sample. Otherwise it belongs to the second sample. 
+
```
What are X and Y here is not clear.

olgarev commented on docs/en/sql-reference/aggregate-functions/reference/mannwhitneyutest.md:
```diff
@@ -0,0 +1,72 @@
+    -   `"less"`;
+    -   if parameter is not specified then default two sided hypothesis is tested.
+-   `sample_data` — sample data. [Float64](../../../sql-reference/data-types/float.md).
+-   `sample_index` — sample index. [Int8](../../../sql-reference/data-types/int-uint.md).
```
```suggestion
-   `sample_index` — sample index. [UInt64](../../../sql-reference/data-types/int-uint.md).
```

lehasm committed:
- First header is first level.
- sample_index type set to UInt64.
- Return values renamed.

- mannWhitneyUTest:
  - added continuity_correction parameter
  - optional parameters wrapped in braces [ ]
  - description refactored
  - example SQL placed on single line

lehasm committed:
Improved formulations (hopefully)
General data type names used (Integer, Float, Decimal)
Removed names from return values

lehasm committed:
Fixes

lehasm committed:
Optional parameters stated explicitly

lehasm committed:
New line at the end of files

lehasm committed:
Russian function description

lehasm committed:
Files moved in the right folder

lehasm committed:
Translated 'Original article' to russian

lehasm committed:
Typos are fixed

* fixed alowed parameter value: "two sided" -> "two-sided"
* duplicate word "require" is removed from exception messages

lehasm committed:
mannWhitneyUTest description changed
Added Tuple link in return values
"samples from two populations" added in short descriptions

lehasm committed:
Russian translation updated

lehasm committed:
fix

lehasm commented:
Noticed in src\AggregateFunctions\AggregateFunctionMannWhitney.h typos in text messages:
* throw Exception("Unknown parameter in aggregate function " + getName() + ". It must be one of: **'two sided'**, 'less', 'greater'", ErrorCodes::BAD_ARGUMENTS); 
Should be two-sided (with dash)
* In several places word require is duplicated: throw Exception("Aggregate function " + getName() + " **require require** second parameter to be a UInt64", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);

lehasm committed:
Merge branch 'alexey-sm-mannwhitney-messages-fix' into alexey-sm-DOCSUP-3908-mannWitneyUTest-welchTTest-studentTTest-doc

lehasm commented:
Fixed source code messages in this pull request
@CurtizJ   Needs 'can be tested' label to test changes

lehasm commented:
May I change PR status from Draft to Ready for review before tests execution?

CurtizJ commented:
BTW, I noticed, that in documentation of most aggregate functions arguments and parameters are described in one section, called "Parameters". But they are semantically different, and in syntax they are setting in two different clauses (two pairs of round braces). I think there should be two sections: arguments and parameters.

For this PR I leaved it as is for consistency.
